tests/test_types.py

```diff
@@ -113,6 +113,22 @@ def test_constrained_bytes_too_long():
     ]
 
 
+def test_constrained_bytes_upper_enabled():
+    class Model(BaseModel):
+        v: conbytes(to_upper=True)
+
+    m = Model(v=b'abcd')
+    assert m.v == b'ABCD'
+
+
+def test_constrained_bytes_upper_disabled():
+    class Model(BaseModel):
+        v: conbytes(to_upper=False)
+
+    m = Model(v=b'aBcD')
+    assert m.v == b'aBcD'
+
+
```

Comment:
We can use `pytest.mark.parametrize` here:

```suggestion
@pytest.mark.parametrize(
    'to_upper, value, result',
    [
        (True, b'abcd', b'ABCD'),
        (True, b'aBcD', b'aBcD'),
    ],
)
def test_constrained_bytes_upper(to_upper, value, result):
    class Model(BaseModel):
        v: conbytes(to_upper=to_upper)

    m = Model(v=value)
    assert m.v == result
```

Also, we can apply it for other tests as well